arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -2035,30 +2043,42 @@ void RocksDBCollection::setupCache() const {
   // by _cacheEnabled already.
   TRI_ASSERT(!ServerState::instance()->isCoordinator());
 
-  if (_cache == nullptr) {
+  auto cache = std::atomic_load_explicit(&_cache, std::memory_order_relaxed);
+  if (cache == nullptr) {
     TRI_ASSERT(_cacheManager != nullptr);
     LOG_TOPIC("f5df2", DEBUG, Logger::CACHE) << "Creating document cache";
-    _cache = _cacheManager->createCache<cache::BinaryKeyHasher>(
+    cache = _cacheManager->createCache<cache::BinaryKeyHasher>(
         cache::CacheType::Transactional);
+
+    std::atomic_store_explicit(&_cache, std::move(cache),
+                               std::memory_order_relaxed);
```

Comment:
If only single thread call setupCache we don't need std::atomic_load_explicit, only store